python/servo/command_base.py

```diff
@@ -612,6 +616,8 @@ def configuration_decorator(self, *args, **kwargs):
                     kwargs.pop('dev', None)
                     kwargs.pop('prod', None)
                     kwargs.pop('profile', None)
+                    self.flavor = kwargs.get('flavor', None)
```

Comment:
I don't think this is the correct fix for the `smoketest` command issue. Removing the `flavor` from kwargs here will mean it never gets passed to the underlying `install`, `package` etc commands. 
i.e I suspect the below command will always behave as if flavor is None and hence the "default" flavor will be used
```
./mach install -r -ohos --flavor=hos
```

Also, we store the flavor value in `self.flavor` but we don't read that anywhere else.

Reply:
> I suspect the below command will always behave as if flavor is None and hence the "default" flavor will be used

That indeed seems to be the case. Sorry, I should have tested that.

> Also, we store the flavor value in self.flavor but we don't read that anywhere else.

Ah, that seems to be a leftover from a different approach.

> I think it make sense to have **kwargs or even just add flavor=None in the smoketest method. The latter is more explicit as it means flavor is logically supported for smoketest (we might want to test the different flavors) but currently not used

I'm not sure if we already smoketest on android - but I guess the question would be, should `smoketest` / `run` install first (which would require knowing the flavor), or just assume that we already installed servo previously? Installing takes a bit of time, so I guess we might want to avoid installing unnecessarily.

Reply:
> I'm not sure if we already smoketest on android

It is [currently disabled](https://github.com/servo/servo/blob/a97afebdcc7a06201638c45a9c7e4b7aab25203a/.github/workflows/android.yml#L100), but the blockers have been addressed, so we should ideally do it.

> Installing takes a bit of time, so I guess we might want to avoid installing unnecessarily.

That is a good point. I guess we can have an `--install` option for such commands and not install by default.